src/org/abymaly/code/magnets/Magnets.java

```diff
@@ -2,4 +2,158 @@
 
 public class Magnets {
 
-}
+    public Signs[][] board;
+    private final int size;
+    public int possibleMoves = 0;
+
+    public Magnets(int size) {
+        this.size = size;
+        this.board = new Signs[size][size];
+    }
+
+    public Magnets() {
+        this.size = 4;
+    }
+
+    class StartPosition {
+        int x;
+        int y;
+
+        public StartPosition() {
+        }
+    }
+
+    StartPosition startPosition = new StartPosition();
+
+    public void printBoard(Signs[][] board) {
+        String newLine = "\n";
+        StringBuffer sbLine = new StringBuffer();
+
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                if (board[i][j].equals(Signs.EMPTY)) {
+                    sbLine.append(" _ ");
+                } else {
+                    sbLine.append(" ");
+                    sbLine.append(board[i][j]);
+                    sbLine.append(" ");
+                }
+            }
+            sbLine.append(newLine);
+        }
+        System.out.println(sbLine);
+    }
+
+    public void setStartPosition(Signs[][] board, int x, int y) {
+        board[x][y] = Signs.O;
+    }
+
+    public void setData(Signs[][] board, int x, int y, Signs value) {
+        board[x][y] = value;
+    }
+
+    public Signs[][] emptyBoard () {
+        Signs [][] newBoard = new Signs[size][size];
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                setData(newBoard, i, j, Signs.EMPTY);
+            }
+        }
+        return newBoard;
+    }
+
+    public void refreshStartPosition(Signs[][] board) {
```

Comment:
Ale ten `startPosition` jest niezainicjowany....
[edit] Już mam. O rany. To nie miało być `startPosition`. To po prostu miało być `x` i `y` określające gdzie aktualnie punkcik jest. Zapakowanie ich w klasę nic nie wnosi.
I chyba słowo "start" jest mylące, bo bym po "start" oczekiwał `final`. Bardziej by mi pasowało "current"